debugtools/DDR_VM/src/com/ibm/j9ddr/vm29/tools/ddrinteractive/commands/DumpModuleExportsCommand.java

```diff
@@ -68,28 +75,15 @@ public void run(String command, String[] args, Context context, PrintStream out)
 		try {
 			J9JavaVMPointer vm = J9RASHelper.getVM(DataType.getJ9RASPointer());
 			if (JavaVersionHelper.ensureJava9AndUp(vm, out)) {
+				GCClassLoaderIterator iterator = GCClassLoaderIterator.from();
 				String targetModuleAddress = args[0];
-				J9PoolPointer pool = vm.modularityPool();
-				Pool<J9ModulePointer> packagePool = Pool.fromJ9Pool(pool, J9ModulePointer.class);
-				SlotIterator<J9ModulePointer> poolIterator = packagePool.iterator();
-				J9ModulePointer ptr = null;
-				J9PackagePointer packagePtr = null;
-				while (poolIterator.hasNext()) {
-					ptr = poolIterator.next();
-					/*
-					 * Since both packages and modules could be in the pool, the
-					 * iterator checks on the region and make sure that it is
-					 * pointing to a package. (For modules, the first field must
-					 * be in heap since it is an object while for j9package it
-					 * is a J9UTF8.)
-					 */
-					if (!ObjectModel.isPointerInHeap(vm, ptr.moduleName())) {
-						packagePtr = J9PackagePointer.cast(ptr);
-						/*
-						 * Iterate through the packages and if the package's
-						 * exported module name matches the target module,
-						 * output its address.
-						 */
+				while (iterator.hasNext()) {
+					J9ClassLoaderPointer classLoaderPointer = iterator.next();
+					HashTable<J9PackagePointer> packageHashTable = PackageHashTable
+							.fromJ9HashTable(classLoaderPointer.packageHashTable());
+					SlotIterator<J9PackagePointer> slotIterator = packageHashTable.iterator();
+					while (slotIterator.hasNext()) {
+						J9PackagePointer packagePtr = slotIterator.next();
```

Comment:
This pattern of iterating class loaders, then packages appears several times: perhaps it would be worth creating an iterator that produced all those packages.

Reply:
This pattern (`package`) only appeared once, the other ones are similar but with `module` . Should I create two separate iterators for each?

Reply:
Sorry, I picked on the wrong file to comment. I thought we had three instances of the same pattern, but, as you point out, only two (with modules in class loaders). Perhaps you could just open an issue to consider capturing that common code.